lib/tasks/expanded_links.rake

```diff
@@ -0,0 +1,73 @@
+namespace :expanded_links do
+  desc "Populate the Expanded Links table in the database"
+  task populate: :environment do
+    document_count = Document.count
+    payload_version = Event.maximum(:id)
+    Document.find_each.with_index do |document, index|
+      update_draft_links(document.content_id, document.locale, payload_version)
+      update_live_links(document.content_id, document.locale, payload_version)
+
+      progress(index, document_count)
+    end
+  end
+
+  desc "
+  Populate the Expanded Links table with content of a particular document type
+  Usage
+  rake 'expanded_links:populate_document_type[:document_type]'
+  "
+  task :populate_document_type, [:document_type] => :environment do |_t, args|
+    tuples = Edition
+      .with_document
+      .where(document_type: args[:document_type])
+      .distinct
+      .pluck("documents.content_id", "documents.locale")
+
+    payload_version = Event.maximum(:id)
+    tuples.each_with_index do |(content_id, locale), index|
+      update_draft_links(content_id, locale, payload_version)
+      update_live_links(content_id, locale, payload_version)
+
+      progress(index, tuples.count, every: 100)
+    end
+  end
+
+
+  def update_draft_links(content_id, locale, payload_version)
```

Comment:
I feel like these two methods could be simplified to reduce duplication by having a single `update_links(content_id, locale, payload_version, with_drafts:)` method.